aiosmtpd/smtp.py

```diff
@@ -329,11 +329,8 @@ def _set_rset_state(self):
         await self.push(status)
 
     async def smtp_NOOP(self, arg):
-        if arg:
-            await self.push('501 Syntax: NOOP')
-        else:
-            status = await self._call_handler_hook('NOOP')
-            await self.push('250 OK' if status is MISSING else status)
+        status = await self._call_handler_hook('NOOP', arg)
```

Comment:
Changing the signature of `handle_NOOP()` is indeed a backward incompatible change, and we *could* deal with that by introspecting the signature of the handler's actual method.  But I'm not sure it's worth it, especially because you'll have to decide what to do if `NOOP` *has* an argument but `handle_NOOP()` doesn't take one (i.e. do you through the arg away, raise an exception, log a warning, etc.?).

How likely is it that there is existing code that includes a `handle_NOOP()`?  I'd say it's pretty unlikely.  How difficult would it be to adjust an existing `handle_NOOP()` method to work with both 1.0 and 1.1?  Not too difficult, since you can add `handle_NOOP(..., arg=None)`.

So, while I don't like breaking APIs in point releases, in this case it's expedient and probably not too dangerous.  With sufficient warning in the NEWS file, I'll say the risk is acceptable.